jumpcloud-ADMU/Powershell/Private/JumpCloudApi/Invoke-SystemPut.ps1

```diff
@@ -0,0 +1,44 @@
+function Invoke-SystemPut {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$jcApiKey,
+        [Parameter(Mandatory = $false)]
+        [string]$jcOrgID,
+        [Parameter(Mandatory = $true)]
+        [string]$systemId,
+        [Parameter(Mandatory = $true)]
+        [object]$Body
+    )
+    $uri = "https://console.jumpcloud.com/api/systems/$systemId"
+
+    $Headers = @{
+        'Accept'       = 'application/json';
+        'Content-Type' = 'application/json';
+        'x-api-key'    = $jcApiKey;
+    }
+    if ($jcOrgID) {
+        $Headers['x-org-id'] = $jcOrgID;
+    }
+    $maxRetries = 3
+    $retryCount = 0
+    do {
+        try {
+            $response = Invoke-RestMethod -Uri $uri -Method Put -Headers $Headers -Body ($Body | ConvertTo-Json)
+            $success = $true
+        } catch {
+            if ($_.Exception.Message -like "*The remote name could not be resolved*") {
```

Comment:
I think it would be safer to reference the exception's error code instead of the message